### PR TITLE
mongo-Ati Takrami

### PR DESCRIPTION
**Pull request by takrami**
https://mongo-bookstore-api.herokuapp.com/

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+app.get("/books/min-rate/:rate", async (req, res) => {
+  try {
+    const { rate } = req.params;
+    if (Number.isNaN(Number(rate))) {
+      throw new Error("rate should be number");
+    }
+    if (rate < 0 || rate > 5) {
+      throw new Error("rate should be between 0 to 5");
+    }
```
Comment (Bellsibub) [inline src/server.js:77]: Nice error handling 😄

File: src/server.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    if (Number.isNaN(Number(id))) {
+      throw new Error("id should be number");
+    }
+    Book.find({ bookID: id })
+      .then((book) => {
+        res.json({ book });
+      })
+      .catch(() => {
+        throw new Error("something went wrong");
+      });
```
Comment (Bellsibub) [inline src/server.js:63]: Would be great to see different error messages here, like "The ID you provided was not found" or something 😉

**Review by Bellsibub — CHANGES_REQUESTED** [review]
Very clean code 👍 and easy to follow and read
However there may be an issue either with Heroku hosting or your mongo dB cluster? Cause the API is not getting any response from the servers. 
This is what happens when I try to get books:
image


So therefore I cannot Approve this request. Hopefully the issue is easily resolved 😉

File: src/models/Book.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import mongoose from "mongoose";
```
Comment (KarolCislowski) [inline src/models/Book.js:1]: Nice that you put your model in the separate file ❤️

File: src/server.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+const port = process.env.PORT;
+const app = express();
+
+app.use(cors());
+app.use(express.json());
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState !== 1) {
+    res.status(500).json({ message: "Internal server error" });
+  } else {
+    next();
```
Comment (KarolCislowski) [inline src/server.js:24]: Your server is returning this error so probably your API can't connect to the database.

**Review by KarolCislowski — APPROVED** [review]
Please check your DB configuration or connection string in your environment 😸

**Review by KarolCislowski — CHANGES_REQUESTED** [review]
Please check DB connection 😄

File: src/server.js (hunk 4 above)
Comment (takrami) [reply src/server.js:24]: The database connection is resolved. would you please chack it again?

**Review by KarolCislowski — APPROVED** [review]
Nice API 👍🏻

**Pull request closed without merging**